src/java_tools/junitrunner/java/com/google/testing/coverage/JacocoLCOVFormatter.java

```diff
@@ -145,36 +145,35 @@ private void processSourceFile(PrintWriter writer, String sourceFile) {
             }
           }
 
+          int firstLine = srcCoverage.getFirstLine();
+          int lastLine = srcCoverage.getLastLine();
+
           // List branches
-          for (IClassCoverage clsCoverage : sourceToClassCoverage.get(sourceFile).values()) {
-            BranchCoverageDetail detail = branchCoverageDetail.get(clsCoverage.getName());
-            if (detail != null) {
-              for (int line : detail.linesWithBranches()) {
-                int numBranches = detail.getBranches(line);
-                boolean executed = detail.getExecutedBit(line);
-                if (executed) {
-                  for (int branchIdx = 0; branchIdx < numBranches; branchIdx++) {
-                    // We haven't got execution counts for branches; just record if they were hit or
-                    // not.
-                    if (detail.getTakenBit(line, branchIdx)) {
-                      writer.printf("BRDA:%d,%d,%d,%d\n", line, 0, branchIdx, 1); // executed, taken
-                    } else {
-                      writer.printf(
+          for (int line = firstLine; line <= lastLine; line++) {
+            ICounter jacocoBranchCounter = srcCoverage.getLine(line).getBranchCounter();
+            if (jacocoBranchCounter.getStatus() != ICounter.EMPTY) {
+              int numBranches = jacocoBranchCounter.getTotalCount();
+              int numCoveredBranches = jacocoBranchCounter.getCoveredCount();
+              boolean executed = jacocoBranchCounter.getStatus() != ICounter.NOT_COVERED;
+              if (executed) {
+                // We haven't got execution counts for branches; just record if they were hit or
+                // not.
+                for (int branchIdx = 0; branchIdx < numCoveredBranches; branchIdx++) {
```

Comment:
This puts all "covered" branches before all "uncovered" branches. Two executions that cause different branches to be run could not be meaningfully merged.